File: Programming101/week0/is_decreasing.py
def is_decreasing(seq):
    decreasing = True
    for i in range(len(seq) - 1):
        if seq[i] <= seq[i+1]:
            decreasing = False
    return decreasing


def main():
    print(is_decreasing([5, 4, 3, 2, 1]))
    print(is_decreasing([1, 2, 3]))
    print(is_decreasing([100, 50, 20]))
    print(is_decreasing([1, 1, 1, 1]))


if __name__ == '__main__':
    main()

File: FrontendJavaScript/week0/2-A-Set-Of-Problems/set1/isDecreasing.js
"use strict";

var isDecreasing = function(seq) {
  return seq.every(function(element, index) {
    if (index+1 === seq.length) {
      return true;
    }
    return seq[index] > seq[index+1];
  });
};

console.log(isDecreasing([5,4,3,2,1]) === true);
console.log(isDecreasing([1,2,3]) === false);
console.log(isDecreasing([100, 50, 20]) === true);
console.log(isDecreasing([1,1,1,1]) === false);
